app/frontend/src/utils/transformUtils.js

```diff
@@ -64,3 +65,147 @@ export function attachAttributesToLinks(formSchemaComponents) {
+ * @param {Object[]} substartDate An object of Moment JS date
+ * @param {Integer} term An integer of number of Days/Weeks OR Years
+ * @param {String} interval A string of days,Weeks,months
+ * @param {Integer} allowLateTerm An integer of number of Days/Weeks OR Years
+ * @param {String} allowLateInterval A string of days,Weeks,months
+ * @param {Object[]} repeatUntil An object of Moment JS date
+ * @param {String} scheduleType A string one of Manual, ClosingDate OR Period
+ * @param {Object[]} closeDate An object of Moment JS date
+ * @returns {Object[]} An object array of Available dates in given period
+ */
+export function getAvailableDates(
+  keepAliveFor=0,
+  keepAliveForInterval='days',
+  submstartDate,
+  term=null,
+  interval=null,
+  allowLateTerm=null,
+  allowLateInterval=null,
+  repeatUntil,
+  scheduleType,
+  closeDate=null
+) {
+
+  let substartDate = moment(submstartDate);
+  repeatUntil = moment(repeatUntil);
+  var calculatedsubcloseDate = getCalculatedCloseSubmissionDate(substartDate,keepAliveFor,keepAliveForInterval,allowLateTerm,allowLateInterval,term,interval,repeatUntil,scheduleType,closeDate);
+  var availableDates = [];
+  if(calculatedsubcloseDate && term && interval) {
+    while (substartDate.isBefore(calculatedsubcloseDate)) {
+      var newDate = substartDate.clone();
+      if(substartDate.isBefore(repeatUntil)){
+        availableDates.push(Object({
+          startDate:substartDate.format('YYYY-MM-DD HH:MM:SS'),
+          closeDate:newDate.add(keepAliveFor,keepAliveForInterval).format('YYYY-MM-DD HH:MM:SS'),
+          graceDate:newDate.add(allowLateTerm,allowLateInterval).format('YYYY-MM-DD HH:MM:SS')
+        }));
+      }
+      substartDate.add(term,interval);
+    }
+  }
+
+  if((term == null && interval == null) && (keepAliveFor && keepAliveForInterval)){
+    var newDates = substartDate.clone();
+    availableDates.push(Object({
+      startDate:substartDate.format('YYYY-MM-DD HH:MM:SS'),
+      closeDate:newDates.add(keepAliveFor,keepAliveForInterval).format('YYYY-MM-DD HH:MM:SS'),
+      graceDate: allowLateTerm && allowLateInterval ? newDates.add(allowLateTerm,allowLateInterval).format('YYYY-MM-DD HH:MM:SS') : null
+    }));
+  }
+  return availableDates;
+}
+
+
+
+/**
+ * @function getCalculatedCloseSubmissionDate
+ * Get calculated Close date for a Form schedule setting with the given scenario
+ *
+ * @param {Object[]} openDate An object of Moment JS date
+ * keepOpenForTerm
+ * keepOpenForInterval
+ * @param {Integer} term An integer of number of Days/Weeks OR Years
+ * @param {String} interval A string of days,Weeks,months
+ * @param {Integer} allowLateTerm An integer of number of Days/Weeks OR Years
+ * @param {String} allowLateInterval A string of days,Weeks,months
+ * @param {Integer} repeatSubmissionTerm An integer of number of Days/Weeks OR Years
+ * @param {String} repeatSubmissionInterval A string of days,Weeks,months
+ * @param {Object[]} repeatUntil An object of Moment JS date
```

Comment:
Why is this called repeatUntil when the declaration is called repeatSubmissionUntil?